### PR TITLE
bump-cask-pr: avoid replacing sha256 twice on multilang cask

### PR DESCRIPTION
**Pull request by xxyzz**
this script always get `inreplace failed` on cask like firefox because the
default sha256 is replaced twice

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/dev-cmd/bump-cask-pr.rb [hunk 1]
```diff
@@ -153,7 +153,7 @@ def bump_cask_pr
       end
     end
 
-    if new_hash.present?
+    if new_hash.present? && cask.language.blank?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-cask-pr.rb:156]: Can you add a comment here explaining why `cask.language` needs to be blank here?

**Review by miccal — APPROVED** [review]

**Review by Rylan12 — COMMENTED** [review]
This doesn't feel like the right solution, here. Line 148 (just above this) is supposed to skip the current language, so this shouldn't be necessary.

It looks like the problem in this case is that the firefox language block is that `cask.laguages` returns an array with `en` as the language code while `cask.language` returns `en-US` or `en-US` (that's what happens to me here in the US and I'm assuming is what's happening for you, too)

I think the better solution is to patch that statement to correctly ignore the right language, even if there's this type of mismatch.

---

Can someone from @Homebrew/cask confirm that my understanding of what's happening in this situation is correct?

Say we have this language declaration:

```ruby
  language "en", default: true do
    sha256 "c439ebf7a75f295dbb1daa8a7488cf87e356fc3dacfa0ce13d6425119fdb621f"
    "en-US"
  end
```

Then `"en"` passed to `language` means that this language should be used for any `en` locale that's not matched to a more specific `en-XY` language that's defined elsewhere in the cask. The `"en-US"` string that's returned clarifies that the default `en` language that will be used is `en-US` and not some other `en` language. Is that correct?

In the case I defined, `Cask#languages` returns `["en"]`. Should `Cask#languages`, instead, return the _actual_ languages (i.e. `["en-US"]`) and not the more generic ones? I feel like it makes more sense to want to know the languages that the cask is _actually_ available in. Or, maybe we can have a method for both (or an option to `#languages` to get the behavior that makes sense)

Alternatively, what about a method `Cask#matched_language` that would, in this case, return `"en"` since that's the locale that was matched to select the language.

Thoughts?

**Comment by xxyzz** [comment]
> I think the better solution is to patch that statement to correctly ignore the right language, even if there's this type of mismatch.

Or replace all sha256 values in that loop by deleting the `next if` line?

**Comment by Rylan12** [comment]
> Or replace all sha256 values in that loop by deleting the `next if` line?

Yeah, that certainly would work and is a possible solution. My feeling is that the better solution is to align `Cask#languages` and `Cask#language` because I think it's confusing that they're different (and I totally understand why this particular case went under the radar), but I could certainly be overridden by other cask folks.

**Comment by xxyzz** [comment]
Could someone review this pull request?

**Review by Rylan12 — APPROVED** [review]
Thanks, @xxyzz! Sorry for the delay

**Pull request merged**